crates/codegen/parser/src/rust_lib_code_generator.rs

```diff
@@ -50,130 +50,116 @@ impl CodeGenerator {
             .unwrap();
 
         {
-            let language_boilerplate_common = codegen
-                .read_file(
-                    &codegen
-                        .repo_root
-                        .join("crates/codegen/parser_templates/src/shared/language.rs"),
-                )
-                .unwrap();
-            let version_flag_declarations = self.version_flag_declarations().join(",");
-            let version_flag_initializers = self.version_flag_initializers().join(",");
-            let scanner_invocations = self.scanner_invocations().join(",");
-            let parser_invocations = self.parser_invocations().join(",");
-            let versions_array = {
-                let versions = grammar.versions.iter().map(|v| v.to_string());
-                quote! { static VERSIONS: &'static [&'static str] = &[ #(#versions),* ]; }
-            };
+            // Use `format!` here because the content contains comments, that `quote!` throws away.
+            let content = format!(
+                "
+                {language_boilerplate_common}
+
+                pub struct Language {{
+                    version: Version,
+                    {version_flag_declarations}
+                }}
+
+                #[derive(thiserror::Error, Debug)]
+                pub enum Error {{
+                    #[error(\"Invalid {grammar_title} language version '{{0}}'.\")]
+                    InvalidLanguageVersion(Version),
+                }}
+
+                impl Language {{
+                    pub fn new(version: Version) -> Result<Self, Error> {{
```

Comment:
I wonder if these can move to `parser_templates` as well?
Not blocking for this PR though.

Reply:
These are either parametric, or language specific, so I don't think so.